[DatePickerThemeData] Add Subheader Color Customization

AffanShaikhsurab opened the pull request:
This PR addresses issue https://github.com/flutter/flutter/issues/160591(https://github.com/flutter/flutter/issues/160591) by adding support for subheader color customization in `DatePickerThemeData`. Previously, it was not possible to directly modify the subheader styling without relying on `onSurface` color in the `colorScheme`.  

#### **Changes**  
- Added a new property `subheaderColor` to `DatePickerThemeData`.  
- Updated the `calendar_date_picker.dart` implementation to use `DatePickerThemeData.subheaderColor` if defined, falling back to the previous behavior if null.  

#### **Code Behavior**  
- If `subheaderColor` is set, it overrides the default styling of the subheader.  
- Fallback logic ensures backward compatibility by using `Theme.of(context).colorScheme.onSurface.withOpacity(0.60)` when `subheaderColor` is not specified.  




<!--
Thanks for filing a pull request!
Reviewers are typically assigned within a week of filing a request.
To learn more about code review, see our documentation on Tree Hygiene: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
-->

### **Before**
Using   `colorScheme: ColorScheme.light(onSurface: Colors.orange),`

![image](https://github.com/user-attachments/assets/f737b093-563a-43a3-b666-9542dc1a2229)


### **After**
Using `subheaderColor :  Colors.orange ,
` in DatePickerThemeData
![image](https://github.com/user-attachments/assets/3451ff53-8d2e-4b9e-9684-db78090adddd)

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I signed the [CLA].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

<!-- Links -->
[Contributor Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#overview
[Tree Hygiene]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md
[test-exempt]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md
[Features we expect every widget to implement]: https://github.com/flutter/flutter/blob/main/docs/contributing/Style-guide-for-Flutter-repo.md#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/blob/main/docs/contributing/Chat.md
[Data Driven Fixes]: https://github.com/flutter/flutter/blob/main/docs/contributing/Data-driven-Fixes.md

AffanShaikhsurab committed:
This PR addresses issue #160591 by adding support for subheader color
customization in DatePickerThemeData. Previously,
it was not possible to directly modify the subheader styling without
 relying on onSurface color in the colorScheme.

maheshj01 commented:
@AffanShaikhsurab Please complete the prelaunch checklist by checking that apply

- [ ] to [X]

And wait for someone to review. To escalate the review process I would suggest reaching out to flutter team members on discord

AffanShaikhsurab committed:
Merge branch 'master' into ui/dartpicker-subheader-color

AffanShaikhsurab commented:
> @AffanShaikhsurab Please complete the prelaunch checklist by checking that apply
> 
> * [ ]  to [X]
> 
> I am not the right person to review. To escalate the review process I would suggest reaching out to flutter team members on discord.

yeah sure , are there any specifc members i should approch too !!?

maheshj01 commented:
Just ask if someone can give feedback on this PR and you should have the right person taking a look. Reviewers are busy with other issues, so someone should take a look at this one as soon as they can.

AffanShaikhsurab commented:
> ou may find reviewers by looking at issues/PRs with similar tags material design, framework

Sure thanks a lot

nate-thegrate commented:
One of the pre-launch checklist items:

> - [x] I read the [Tree Hygiene](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md) wiki page, which explains my responsibilities.

<br>

For future reference, that page recommends waiting 2 weeks before reaching out on Discord, to allow time for the contribution to go through the normal PR triage process.

AffanShaikhsurab commented:
> One of the pre-launch checklist items:
> 
> > * [x]  I read the [Tree Hygiene](https://github.com/flutter/flutter/blob/main/docs/contributing/Tree-hygiene.md) wiki page, which explains my responsibilities.
> 
> 
> For future reference, that page recommends waiting 2 weeks before reaching out on Discord, to allow time for the contribution to go through the normal PR triage process

I will make sure of it from now on

AffanShaikhsurab committed:
Update calendar_date_picker.dart

removed blank line . as mentioned in the error The error messages reported above are repeated here:
  -- This line intentionally left blank --  
/b/s/w/ir/x/w/flutter/packages/flutter/lib/src/material/date_picker_theme.dart:1164: trailing U+0020 space character in Linux analyze

AffanShaikhsurab committed:
Merge branch 'master' into ui/dartpicker-subheader-color

AffanShaikhsurab committed:
dart formatted date_picker_theme.dart

AffanShaikhsurab committed:
reverted date_picker_theme.dart

QuncCccccc reviewed (commented):
Thanks so much for your contribution! There's a similar PR https://github.com/flutter/flutter/pull/161458 and that one seems provide more flexibility that we can customize the text button and the toggle button colors separately. Would you mind if I review that one first?

AffanShaikhsurab commented:
> Thanks so much for your contribution! There's a similar PR #161458 and that one seems provide more flexibility that we can customize the text button and the toggle button colors separately. Would you mind if I review that one first?

Yeah, sure! If that offers a better approach, then it's definitely better.

AffanShaikhsurab committed:
Merge branch 'master' into ui/dartpicker-subheader-color

AffanShaikhsurab committed:
Merge branch 'master' into ui/dartpicker-subheader-color

AffanShaikhsurab committed:
Merge branch 'master' into ui/dartpicker-subheader-color

QuncCccccc commented:
Thanks so much for your understanding:)

Piinks commented:
Per the conversation here, I am going to close this PR. Thank you for contributing @AffanShaikhsurab! It is much appreciated. :)
